### PR TITLE
Update ArrayExtensions.swift

### PR DESCRIPTION
**Pull request by fallwd**
For creating a dictionary.Dictionary(uniqueKeysWithValues:) is more efficient than reduce(into:) to. i used time profile  and found a significant improvement.

## Checklist

- [ x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [ x] New extensions are written in Swift 5.6.
- [ x] New extensions support iOS 12.0+ / tvOS 12.0+ / macOS 10.13+ / watchOS 4.0+, or use `@available` if not.
- [ ] I have added tests for new extensions, and they passed.
- [ ] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [ ] All extensions are declared as **public**.
- [ ] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

**Comment by fallwd** [comment]
let items = (1...1000000).map { String($0) }

func createDictUsingUniqueKeysWithValues() {
    _ = Dictionary(uniqueKeysWithValues: items.enumerated().map { ($1, $0) })
}

func createDictUsingReduce() {
    _ = items.enumerated().reduce(into: [:]) { $0[$1.element] = $1.offset }
}

time profile result:
items = (1...1000000).map { String($0) } 
createDictUsingReduce()  125ms
createDictUsingUniqueKeysWithValues() 63ms

items = (1...100000000).map { String($0) }
createDictUsingReduce()  20.26s
createDictUsingUniqueKeysWithValues() 12.58s

**Comment by fallwd** [comment]
Darwin error:
/Users/runner/work/SwifterSwift/SwifterSwift/Sources/SwifterSwift/Foundation/NSAttributedStringExtensions.swift:112:63: error: cannot find 'pattern' in scope
        guard let pattern = try? NSRegularExpression(pattern: pattern, options: options) else { return self }

File: Sources/SwifterSwift/SwiftStdlib/ArrayExtensions.swift [hunk 1]
```diff
@@ -58,7 +58,7 @@ public extension Array {
     ///   - keyPath: keyPath indicating the property that the array should be sorted by
     /// - Returns: sorted array.
     func sorted<T: Hashable>(like otherArray: [T], keyPath: KeyPath<Element, T>) -> [Element] {
-        let dict = otherArray.enumerated().reduce(into: [:]) { $0[$1.element] = $1.offset }
+        let dict = Dictionary(uniqueKeysWithValues: otherArray.enumerated().map { ($1, $0) })
```
Comment (guykogus) [inline Sources/SwifterSwift/SwiftStdlib/ArrayExtensions.swift:61]: Could you use `lazy` here? Would it speed things up even more?
```suggestion
        let dict = Dictionary(uniqueKeysWithValues: otherArray.enumerated().lazy.map { ($1, $0) })
```

Comment (guykogus) [reply Sources/SwifterSwift/SwiftStdlib/ArrayExtensions.swift:61]: Tested it myself, it makes virtually no difference.

**Pull request merged**